calendar-scheduling/src/main/java/com/linagora/calendar/scheduling/AlarmTriggerService.java

```diff
@@ -0,0 +1,248 @@
+                    .flatMap(alarmEventDAO::update)
+                    .then())
+                .orElse(alarmEventDAO.delete(alarmEvent.eventUid(), alarmEvent.recipient()));
+        }
+        return alarmEventDAO.delete(alarmEvent.eventUid(), alarmEvent.recipient());
+    }
+
+    private Mono<Void> sendMail(AlarmEvent alarmEvent, Instant now) {
+        Username recipientUser = Username.fromMailAddress(alarmEvent.recipient());
+        return getUserSettings(recipientUser).flatMap(resolvedSettings -> {
+            Locale locale = resolvedSettings.locale();
+            var model = toPugModel(CalendarUtil.parseIcs(alarmEvent.ics()),
+                alarmEvent.recurrenceId(),
+                locale,
+                Duration.between(now, alarmEvent.eventStartTime()));
+            return Mono.fromCallable(() -> messageGeneratorFactory.forLocalizedFeature(new Language(locale), TEMPLATE_TYPE))
+                .flatMap(messageGenerator -> messageGenerator.generate(recipientUser, maybeSender.asOptional(), model, List.of()))
+                .flatMap(message -> mailSenderFactory.create()
+                    .flatMap(mailSender -> mailSender.send(new Mail(maybeSender, List.of(alarmEvent.recipient()), message))));
+        });
+    }
+
+    private Mono<ResolvedSettings> getUserSettings(Username user) {
+        return settingsBasedResolver.readSavedSettings(sessionProvider.createSession(user))
+            .defaultIfEmpty(ResolvedSettings.DEFAULT)
+            .doOnError(error -> {
+                if (!(error instanceof DomainNotFoundException)) {
+                    LOGGER.error("Error resolving user settings for {}, will use default settings: {}",
+                        user.asString(), ResolvedSettings.DEFAULT, error);
+                }
+            }).onErrorResume(error -> Mono.just(ResolvedSettings.DEFAULT));
+    }
+
+    private Map<String, Object> toPugModel(Calendar calendar,
+                                           Optional<String> maybeRecurrenceId,
+                                           Locale locale,
+                                           Duration duration) {
+        VEvent vEvent = maybeRecurrenceId.map(recurrenceId -> getVEvent(calendar, recurrenceId))
+            .orElse(GET_FIRST_VEVENT_FUNCTION.apply(calendar));
+        PersonModel organizer = PERSON_TO_MODEL.apply(EventParseUtils.getOrganizer(vEvent));
+        String summary = EventParseUtils.getSummary(vEvent).orElse(StringUtils.EMPTY);
+
+        ImmutableMap.Builder<String, Object> contentBuilder = ImmutableMap.builder();
+        contentBuilder.put("event", toPugModel(vEvent))
+            .put("duration", formatDuration(duration, locale));
+
+        return ImmutableMap.of("content", contentBuilder.build(),
+            "subject.summary", summary,
+            "subject.organizer", StringUtils.defaultIfEmpty(organizer.cn(), organizer.email()));
```

Comment:
If the organizer’s CN name is empty, please query `UserDAO` to retrieve it, similar to our approach in the event email.